util/collate/charset.go

```diff
@@ -23,6 +23,6 @@ func switchDefaultCollation(flag bool) {
 	} else {
 		charset.CharacterSetInfos[charset.CharsetGBK].DefaultCollation = charset.CollationGBKBin
 	}
-	charset.CharacterSetInfos[charset.CharsetGBK].Collations[charset.CollationGBKBin].IsDefault = !flag
-	charset.CharacterSetInfos[charset.CharsetGBK].Collations[charset.CollationGBKChineseCI].IsDefault = flag
+	charset.CharacterSetInfos[charset.CharsetGBK].Collations[charset.CollationGBKBin].IsDefault.Store(!flag)
```

Comment:
I think `switchDefaultCollation` is only called in test, so the data race only appears in test. Should we change a lot to fix this, not sure for me.

Reply:
it may be related with 

https://github.com/pingcap/tidb/blob/1fac141f35446c8942e2f413afbe2cd54cdb0964/executor/point_get_test.go#L800

Reply:
All the data race goroutine starts with L800. We can not run parallelly test or set atomic for this value.

Reply:
or move `wg.Wait()` into for loop

Reply:
I have removed the outside ```wg.Wait()```  when to call ```testLock```.